_ml-commons-plugin/api.md

```diff
@@ -115,6 +115,120 @@ The API returns information on the model, the algorithm used, and the content fo
 }
 ```
 
+## Upload load a model
+
+Use the upload operation to upload a pre-existing or custom model and into a model index. ML commons splits the model into smaller chunks and saves those chunks into the model's index.
+
+```json
+POST /_plugins/_ml/models/_upload
+```
+
+### Request fields
+
+Field | Data Type | Description
+:---  | :--- | :--- 
+`name`| string | The name of the model. |
+`version` | integer | The version number of the model. Version number should follow standard semantic format, major.minor.patch |
+`model_format` | string | The portable format of the model file. Currently only supports `TORCH_SCRIPT`. |
+`model_config` _(Optional)_ | string | The model's configuration, including the `model_type`, `embedding_dimension`, and `framework_type`. |
+`url` | string | The URL of the server which contains the model. |
```

Comment:
The model URL?

Reply:
Yeah, I think more clarity is required to explain what this URL is. The URL specifies the location of the model artifacts. For instance, this might be a download URL for the model on a web server or Amazon S3. Invoking this API will instruct our OpenSearch server to pull the model artifacts from the URL.